### ISSUE TITLE
Typescript 4 warning

### ISSUE THREAD

**Issue by nojvek** [typescript-eslint/typescript-eslint#2411]
Typescript 4 was released but typescript-eslint/parser (3.9.1) gives this warning

```
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <3.10.0

YOUR TYPESCRIPT VERSION: 4.0.2

Please only submit bug reports when using the officially supported version.

=============
```

**Comment by nojvek** [comment]
For what it's worth, It worked fine with typescript 4

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/issues/2260
